sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +764,83 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Initiates a search and returns a {@link PhoneNumberSearch} usable by other functions
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param options A {@link CreateSearchOptions} with the search options
+     * @param pollInterval The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the search result
+     */
+    public PollerFlux<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
+        CreateSearchOptions options, Duration pollInterval) {
+        Objects.requireNonNull(options, "'options' cannot be null.");
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(pollInterval,
+            createSearchActivationOperation(options),
+            createSearchPollOperation(),
+            cancelSearchOperation(),
+            createSearchFetchResultOperation());
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PhoneNumberSearch>> createSearchActivationOperation(CreateSearchOptions options) {
```

Comment:
Sorry I should have mentioned it in the previous review iteration but this style of specifying parameters in different lines is quite confusing, did you see this as a standard somewhere? Shouldn't this be better --

private Function<PollingContext<PhoneNumberSearch>, Mono<PhoneNumberSearch>> 
      createSearchActivationOperation(CreateSearchOptions options) {